src/sbvr-api/permissions.ts

```diff
@@ -684,6 +684,182 @@ const getAlias = (name: string) => {
 	}
 	return `permissions${permissionsJSON}`;
 };
+
+const rewriteRelationship = memoizeWeak(
+	(
+		value: Relationship,
+		name: string,
+		abstractSqlModel: AbstractSqlModel,
+		permissionsLookup: PermissionLookup,
+		vocabulary: string,
+	) => {
+		let escapedName = sqlNameToODataName(name);
+		if (abstractSqlModel.tables[name]) {
+			escapedName = sqlNameToODataName(abstractSqlModel.tables[name].name);
+		}
+
+		const originalAbstractSQLModel = sbvrUtils.getAbstractSqlModel({
+			vocabulary,
+		});
+
+		const rewrite = (object: AnyObject) => {
```

Comment:
Ah, based on the comments I can now see this should be
```suggestion
		const rewrite = (object: Relationship | RelationshipMapping) => {
```
and then the rest of the types should naturally fall out of it